### PR TITLE
Support nested observation in critic_network_visualizer

### PR DESCRIPTION
**Pull request by Haichao-Zhang**

### CODE REVIEW

File: alf/utils/visualizer.py [hunk 1]
```diff
@@ -86,17 +88,22 @@ def net_func(net_input):
         W (int): number of samples to be used for creating visualization along
             the direction of ``action_upper_right - action_upper_left``.
             The total number of samples is H * W.
+        batch_size (int): the batch size of the input ``observation``. If None,
```
Comment (hnyu) [inline alf/utils/visualizer.py:91]: If None, ...?

File: alf/utils/visualizer.py [hunk 2]
```diff
@@ -86,17 +88,22 @@ def net_func(net_input):
...
+
 
     Returns:
         The network response image of the shape [B, K, H, W], where K denotes
         the dimensionality of the network output for the non-batch dimension.
     """
+    if batch_size is None:
+        batch_size = nest.flatten(observation)[0].shape[0]
```
Comment (hnyu) [inline alf/utils/visualizer.py]: Can use `get_nest_size()`

**Review by hnyu — CHANGES_REQUESTED** [review]

File: alf/utils/visualizer.py (hunk 2 above)
Comment (Haichao-Zhang) [reply alf/utils/visualizer.py]: Changed

File: alf/utils/visualizer.py (hunk 1 above)
Comment (Haichao-Zhang) [reply alf/utils/visualizer.py:91]: Completed the sentence.

**Review by hnyu — APPROVED** [review]

**Pull request merged**